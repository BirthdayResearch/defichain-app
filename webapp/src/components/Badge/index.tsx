import React from 'react';
import { MdInfo } from 'react-icons/md';
import { I18n } from 'react-redux-i18n';
import { Button, ButtonProps } from 'reactstrap';

interface BadgeProps extends ButtonProps {
  baseClass?: string;
  label: string;
}

const Badge: React.FunctionComponent<BadgeProps> = (props: BadgeProps) => {
  return (
    <div className={props.baseClass}>
      <Button className='badge' {...props}>
        <MdInfo />
        <span>{props.label}</span>
      </Button>
    </div>
  );
};

export default Badge;
