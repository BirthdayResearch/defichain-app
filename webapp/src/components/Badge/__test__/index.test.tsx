import React from 'react';
import Badge from '../index';
import { shallow } from 'enzyme';

describe('Badge component', () => {
  it('should check for snapshot', () => {
    const props = {
      baseClass: 'baseClass',
      label: 'label',
    };

    const wrapper = shallow(<Badge {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
