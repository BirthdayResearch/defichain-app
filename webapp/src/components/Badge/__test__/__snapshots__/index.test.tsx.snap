// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component should check for snapshot 1`] = `
<div
  className="baseClass"
>
  <Button
    baseClass="baseClass"
    className="badge"
    color="secondary"
    label="label"
    tag="button"
  >
    <MdInfo />
    <span>
      label
    </span>
  </Button>
</div>
`;
